Quadriga/src/main/java/edu/asu/spring/quadriga/service/impl/UserManager.java

```diff
@@ -458,5 +511,63 @@ public IUserFactory getUserFactory() {
     public void setUserFactory(IUserFactory userFactory) {
         this.userFactory = userFactory;
     }
-
+    
+    public String getUniqueUsername(String providerId) {
+        String id = null;
+        while (true) {
+            id = "USR" + generateUniqueId() + "_" + providerId;
+            Object existingFile = null;
+            try {
+                existingFile = getUser(id);
+            } catch (QuadrigaStorageException e) {
+                // TODO Auto-generated catch block
+                e.printStackTrace();
+            }
+            if (existingFile == null) {
+                break;
+            }
+        }
+        return id;
+    }
+    
+    /**
+     * This method is used to search the user based on the userId and the social provider that authenticates the user (e.g. github, facebook etc.)
+     * @param userId  user's id in the social provider e.g. username in github.
+     * @param provider social provider used for user authentication (e.g. github, facebook etc.)
+     * @return user information
+     * @throws QuadrigaStorageException
+     * @author chiraag subramanian
+     */
+   public IUser findUserByProviderUserId(String userId, String provider) throws QuadrigaStorageException {
+       try {
+           return userDeepMapper.findUserByProviderUserId(userId, provider);
+       } catch (Exception ex) {
+           throw new QuadrigaStorageException();
```

Comment:
several things:
- never catch Exception unless it's explicitly thrown (always be specific)
- why are you catching a QuadrigaStorageException if all you do is throw a new QuadrigaStorageException?
- NEVER EVER swallow exceptions (once this is thrown, nobody can see from the log what the actual issue was because you just throw ex away).